test/float16.jl

```diff
@@ -14,6 +16,25 @@ g = Float16(1.)
 @test isless(g, f)
 @test !isless(f, g)
 
+@test convert(Bool,Float16(0.)) == false
+@test convert(Bool,Float16(0.1)) == true
```

Comment:
Missed from #10748 I guess. Yay, test identifying easy-fix bugs. Looks like `BigFloat` may have the same problem, as does `doc/manual/conversion-and-promotion.rst`

```
tkelman@ygdesk:~/Julia/julia$ git grep -n 'convert(::Type{Bool}'
base/bool.jl:5:convert(::Type{Bool}, x::Bool) = x
base/bool.jl:6:convert(::Type{Bool}, x::Real) = x==0 ? false : x==1 ? true : throw(InexactError())
base/float16.jl:101:convert(::Type{Bool},    x::Float16) = (x!=0)
base/mpfr.jl:159:convert(::Type{Bool}, x::BigFloat) = (x != 0)
base/rational.jl:66:convert(::Type{Bool}, x::Rational) = x==0 ? false : x==1 ? true : throw(InexactE
doc/manual/conversion-and-promotion.rst:124:    convert(::Type{Bool}, x::Number) = (x!=0)
```

Hopefully nothing else?